### PR TITLE
docs: Add guide for album sync python script

### PR DESCRIPTION
**Pull request by markeeisner**
I created a python script to sync a single album to a local folder for my digital photoframe project. I thought it might be useful for other users. In discussion #7735 @mmomjian suggested the best way to share the script was with a guide.

This guide briefly shows installation and usage. It links to the original repo for the script (my self hosted gitlab server) but also includes the full example config.ini and the script itself in it's current form.

### CODE REVIEW

**Comment by bo0tzz** [comment]
Thanks for the PR! I'm not sure whether we should set the precedent of adding arbitrary scripts to our documentation. The python file upload is there only to serve as a rudimentary example, and scripts meant for actual usage are much better kept in actual version control. 

An alternative that I would be partial to is to have something like a "Community" section in the documentation where we can link to external tools like [immich-go](https://github.com/simulot/immich-go) and [ImmichFrame](https://github.com/3rob3/ImmichFrame), as well as to scripts like yours and other ones posted in the github discussions.

**Comment by markeeisner** [comment]
> Thanks for the PR! I'm not sure whether we should set the precedent of adding arbitrary scripts to our documentation. The python file upload is there only to serve as a rudimentary example, and scripts meant for actual usage are much better kept in actual version control.
> 
> An alternative that I would be partial to is to have something like a "Community" section in the documentation where we can link to external tools like [immich-go](https://github.com/simulot/immich-go) and [ImmichFrame](https://github.com/3rob3/ImmichFrame), as well as to scripts like yours and other ones posted in the github discussions.

My original thought was that random python scripts weren't suitable for the repo.

An option suggested in #7735 was linking to external tools/examples in a curated list as you suggest. I still think there is value to more practical examples of python/curl api usage in the main documentation alongside the existing 'raw' API documentation but that could be done more neatly with a bit more effort than throwing random scripts at the wall as this PR does. I would be open to improving this script and creating other examples so there is a small selection of official practical examples under the same style to add to the documentation. We'd just need to decide the scope of examples and keep it small.

I would have appreciated and got running a lot quicker if there were a few more examples along the line of the Python File Upload guide and mine available, even if it is just a curated list of external examples.

The linked repo has an adequete readme so is suitable to be linked to in a curated list if adding this guide or exanding the documentation with more examples is unsuitable.

File: docs/docs/guides/api-album-sync.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+# Download each image from the URL and save it to the directory
+headers = {
+    'Accept': 'application/octet-stream',
+    'x-api-key': apikey
+}
+
+photolist = []
+
+for id in image_urls:
+    url = "https://photo.eisner.uk/api/download/asset/" + str(id['id'])
```
Comment (jrasm91) [inline docs/docs/guides/api-album-sync.md:106]: This still uses your domain. Also it looks like this assumes all files are jpg.

**Review by jrasm91 — APPROVED** [review]
Nice! I'm fine with this. Let's see what others think.

File: docs/docs/guides/api-album-sync.md (hunk 1 above)
Comment (markeeisner) [reply docs/docs/guides/api-album-sync.md:106]: Thanks, I'll fix both. The hardcoded jpg is just something that comes from it starting as a basic script for my use case but I can improve it, especially with helpful critique.

Comment (markeeisner) [reply docs/docs/guides/api-album-sync.md]: I think d5331ead8d72f2e9009693cf888010e5b58e2094 both shows my hacky inexperience and fixes the hardcoded url and extension. The only way I could see to get it was another call to the api for the asset info.

File: docs/docs/guides/api-album-sync.md [hunk 2]
```diff
@@ -103,13 +103,20 @@ headers = {
...
+    response = requests.get(assetinfourl, headers=headers)
+    assetinfo = response.json()
+    ext = os.path.splitext(assetinfo['originalFileName'])
+
+    asseturl = url + "/api/download/asset/" + str(id['id'])
+    response = requests.post(asseturl, headers=headers, stream=True)
+
     # Build current photo list for deletions below
-    photo = os.path.basename(url) + '.jpg'
+    photo = os.path.basename(asseturl) + ext[1]
```
Comment (mmomjian) [inline docs/docs/guides/api-album-sync.md:116]: edit: disregard

File: docs/docs/guides/api-album-sync.md (hunk 1 above)
Comment (jrasm91) [reply docs/docs/guides/api-album-sync.md]: Oh nice. Yeah, maybe we should send it with a content-disposition header which could provide a filename in the download response itself.

File: docs/docs/guides/api-album-sync.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    asseturl = url + "/api/download/asset/" + str(id['id'])
+    response = requests.post(asseturl, headers=headers, stream=True)
+
+    # Build current photo list for deletions below
+    photo = os.path.basename(asseturl) + ext[1]
+    photolist.append(photo)
+
+    photofullpath = photodir + '/' + os.path.basename(asseturl) + ext[1]
+    # Only download file if it doesn't already exist
+    if os.path.exists(photofullpath) == False:
```
Comment (mertalev) [inline docs/docs/guides/api-album-sync.md:121]:
```suggestion
    if not os.path.exists(photofullpath):
```

File: docs/docs/guides/api-album-sync.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+for item in data:
+    if item['albumName'] == albumname:
+        albumid = item['id']
+
+# Get the list of photos from the API using the albumid
+response = requests.get(url + "/api/album/" + albumid, headers=headers)
+
+# Parse the JSON response and extract the URLs of the images
+data = response.json()
+image_urls = [image for image in data['assets']]
```
Comment (mertalev) [inline docs/docs/guides/api-album-sync.md:95]:
```suggestion
image_urls = data['assets']
```

File: docs/docs/guides/api-album-sync.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
+# API Album Sync (Python Script)
+
+This is an example of a python script for syncing an album to a local folder. This was used for a digital photoframe so the displayed photos could be managed from the immich web or app UI.
+
+The script is copied below in it's current form. A repository is hosted [here](https://git.orenit.solutions/open/immichalbumpull).
+
+:::danger
+This guide uses a generated API key. This key gives the same access to your immich instance as the user it is attached to, be careful how the config file is stored and transferred.
```
Comment (mertalev) [inline docs/docs/guides/api-album-sync.md:8]:
```suggestion
This guide uses a generated API key. This key gives the same access to your immich instance as the user it is attached to, so be careful how the config file is stored and transferred.
```

**Review by mertalev — APPROVED** [review]

**Comment by alextran1502** [comment]
@markeeisner Thank you for your contribution; I think we eventually want to point this to a Gist that you are maintaining. Do you have a gist address for this?

**Comment by markeeisner** [comment]
There's no gist for it at the moment. It's primarily hosted on my home gitlab server rather than on GitHub but I'm open to moving things around if there's a master plan for a curated list of similar linked resources.

**Pull request merged**